contracts/escrow_fee_distributor/src/contract.rs

```diff
@@ -308,6 +318,10 @@ fn calc_claim_amount(deps: DepsMut, env: Env, account: Addr, config: Config) ->
             break;
         }
 
+        if max_periods.is_some() && Some(max_period_counter) >= max_periods {
```

Comment:
If the max period is omitted then there is still possible out of gas error. I suggest to replace `loop` with `for claim_period in claim_period..max_period` where `max_period = min(max_periods, current_period - claim_period)`. max_periods should be set to some default if it is None.